website/project/tasks.py

```diff
@@ -44,7 +44,7 @@ def update_collecting_metadata(node, saved_fields):
         if node.is_public:
             update_collected_metadata(node._id)
         else:
-            if 'is_public' in saved_fields:
+            if set(saved_fields).intersection({'spam_status', 'is_deleted', 'deleted', 'is_public'}):
```

Comment:
As mentioned above, I don't see how the spam_status, is_deleted, or deleted properties need to be checked only if `not node.is_public`.